internal/platform/request/request.go

```diff
@@ -0,0 +1,71 @@
+package request
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	stdpath "path"
+
+	"github.com/coralproject/shelf/internal/platform/auth"
+)
+
+// Client contains the necessary pieces to perform requests down to a service
+// layer that has platform authentication enabled.
+type Client struct {
+	BaseURL string
+	Signer  auth.Signer
+}
+
+// New creates a new request sourced from the client. If a signer is present on
+// the client, requests will automatically be signed.
+func (c *Client) New(context interface{}, verb, path string, body io.Reader) (*http.Request, error) {
+
+	// Parse the base url passed in.
+	uri, err := url.Parse(c.BaseURL)
+	if err != nil {
+		return nil, err
+	}
+
+	// Join the paths from the base url and the request path.
+	uri.Path = stdpath.Join(uri.Path, path)
```

Comment:
`stdpath.Join` will ensure that even if the path is empty that it will join the path correctly.